draft-brunstrom-taps-impl.md

```diff
@@ -330,6 +343,32 @@ Protocol options are checked next in order. Whether or not a set of protocol, or
 
 Branching for derived endpoints is the final step, and may have multiple layers of derivation or resolution, such as DNS service resolution and DNS hostname resolution.
 
+
+## Ranking Branches
+
+Implementations SHOULD rank the branches of the tree of connection options in order to prioritize them for candidate racing. Leaf nodes on branches with higher rankings represent connection attempts that will be raced first. Implementations SHOULD order the branches to reflect the preferences expressed by the application for its new connection, including Protocol and Path Selection Properties as well as Application Intents, which are specified in {{draft-trammell-taps-interface}}. In addition to the properties provided by the application, an implementation MAY include additional criteria such as cached performance estimates, see {{performance-caches}}, or system policy, see {{role-of-system-policy}}, in the ranking.
+
+The Application Intents specified in {{draft-trammell-taps-interface}} may be used to rank branches in the following ways:
+
+* Capacity Profile:
+An implementation may use the Capacity Profile to prefer paths optimized for the application's expected traffic pattern:
+   * Interactive/Low Latency:
+     Prefer paths with the lowest expected latency
+   * Constant Rate:
+     Prefer paths that can satisfy the requested Stream Send or Stream Receive Bitrate
+   * Scavenger/Bulk:
+     Prefer paths with the highest expected available bandwidth
+
```

Comment:
For each of these path properties, I'd ask: "how can you know?". Please give an example.

Reply:
We can observe the maximum throughput over time in a performance cache, which we use as an estimate of the available bandwidth. I added some clarification to the parts you marked as well as one sentence to Section 8.2 (Performance Caches).

Reply:
Yes, I think this relates to the cached performance parameters, how the exact calculation of such parameters is done, their cache lifetime, etc. 
I'm currently working on another paper that deals with how to get good performance estimates, and I'm looking forward to discussing this more and perhaps contributing some insights to the draft... perhaps post-London. :)